planner/core/plan_cost_ver2.go

```diff
@@ -595,7 +596,17 @@ func (p *PhysicalIndexJoin) getIndexJoinCostVer2(taskType property.TaskType, opt
 	batchRatio := 6.0
 	probeCost := divCostVer2(mulCostVer2(probeChildCost, buildRows), batchRatio)
 
-	p.planCostVer2 = sumCostVer2(startCost, buildChildCost, buildFilterCost, buildTaskCost, divCostVer2(sumCostVer2(probeCost, probeFilterCost, hashTableCost), probeConcurrency))
+	// Double Read Cost
+	var doubleReadCost costVer2
+	if p.ctx.GetSessionVars().IndexJoinDoubleReadPenaltyCostRate > 0 {
+		batchSize := float64(p.ctx.GetSessionVars().IndexJoinBatchSize)
+		taskPerBatch := 1024.0 // TODO: remove this magic number
```

Comment:
Why this magic number?

Reply:
From the OSS case, this magic number can let the planner select a MPP plan when the `penalty-rate=1`.